distribution/scripts/icp.sh

```
@@ -17,35 +17,187 @@
+
+    if [ -f "$CONFIG_FILE" ]; then
+        echo "Starting ICP Server with configuration: $CONFIG_FILE"
+        export BAL_CONFIG_FILES="$CONFIG_FILE"
+    else
+        echo "Warning: Configuration file not found at $CONFIG_FILE"
+        echo "Starting ICP Server without custom configuration..."
+        unset BAL_CONFIG_FILES
+    fi
+
+    if [ "$mode" = "background" ]; then
+        nohup java $JAVA_OPTS -jar "$JAR_FILE" "$@" >> "$LOG_FILE" 2>&1 &
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)